force-app/main/default/classes/SOQL.cls

```diff
@@ -270,12 +289,10 @@ public with sharing class SOQL {
      * @return List of contacts with the same Mailing State as the Account Billing State
      */
     public static List<Contact> getAccountsWithContactsByState(Id accountId) {
-        // Query the Account
-
-        // Store the Account's BillingState
-
-        // Query the Contacts
-        return null; // Replace null with the variable you used to store the result
+        List<Account> accWithBillingState = [SELECT Id, Name, BillingState FROM Account WHERE Id = :accountId];
+        String accBillingState = accWithBillingState[0].BillingState;
+        List<Contact> contactWityMailingAddress = [SELECT Id, Name, MailingState FROM Contact WHERE MailingState = :accBillingState];
```

Comment:
See if you can combine these into a single query. HINT you can use a query as a filter criteria